crates/ty_python_semantic/src/types/diagnostic.rs

```diff
@@ -666,7 +813,26 @@ declare_lint! {
 }
 
 declare_lint! {
-    /// TODO #14889
+    /// ## What it does
+    /// Checks for type variables that has only one constraint.
+    ///
+    /// ## Why is this bad?
+    /// A type variable must either not constrained,
+    /// or have at least two constraints.
+    ///
+    /// ## Examples
+    /// ```python
+    /// from typing import TypeVar
+    ///
+    /// T = TypeVar('T', str)
+    /// ```
+    ///
+    /// Use instead:
+    /// ```python
+    /// T = TypeVar('T', str, int)
+    /// # or
+    /// T = TypeVar('T', bound=str)
```

Comment:
```suggestion
    /// T = TypeVar('T', str)  # invalid constrained TypeVar
    /// ```
    ///
    /// Use instead:
    /// ```python
    /// T = TypeVar('T', str, int)  # valid constrained TypeVar
    /// # or
    /// T = TypeVar('T', bound=str)  # valid bound TypeVar
```